### PR TITLE
Fix scaling for iPhones

### PR DESCRIPTION
**Pull request by wutschel**
This fixes issue https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/111.

Details already given in https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/99#issuecomment-757415645:

> Still not checked for all screens and view modes. It is hard to see what exactly was calculated to get those hard coded numbers. In general the code on master tries to use different scaling factors for several items (e.g. thumbviews, the remote or the volume slider). There are three families of scaling factors which reflect the three different x-axis resolutions at the time when the last changes were done, the fourth one (e.g. iPhone 12) was not taken into account. As the factors are not 100% identical for each family (like 1.29, 1.294 and 1.31) I think there was some manual tweaking done in some places, possibly to handle the effect of separators.
Also, I doubt that all of the code is still active. In some places I could mess around with the scaling and the thumbviews and remote still looked fine.

This PR implement a centralized macro/define which calculates the transform/scaling factor from the current screen width. This factor is now applied instead of hard coded values. The macros should support future screen sizes as well.

Test were done on simulator with
- iPhone 12 Pro Max (width = 428)
- iPhone 11 Pro Max (width = 414)
- iPhone Xs (width = 375)
- iPod Touch (width = 320) and 
- iPad 8G (only to double check if regressions are seen for non-iPhone devices).

As I am mainly using Music it would be good to have some tests specifically with Movies and TV Shows.

### CODE REVIEW

File: XBMC Remote/NowPlaying.m [hunk 1]
```diff
@@ -382,12 +382,20 @@ - (UIImage*)imageWithBorderFromImage:(UIImage*)source{
 int storedItemID;
 int currentItemID;
 
+-(CGRect)transformFrame:(CGRect)frame{
+    CGFloat center_x = frame.origin.x + frame.size.width/2;
+    CGFloat center_y = frame.origin.y + frame.size.height/2;
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:387]: there're `CGRectGetMid...` functions ;)

File: XBMC Remote/RemoteController.m [hunk 2]
```diff
@@ -53,20 +53,21 @@ -(void)hideButton:(NSArray *)buttonsToDo hide:(BOOL)hide{
...
-        transform = 1.16f;
+    if (transform>1.0f && transform<1.29f) {
+        // All devices with 320 > width > 414
         startX = 3;
         transViewY = 58;
     }
-    else if (IS_IPHONE_6_PLUS){
-        transform = 1.29f;
+    else if (transform>=1.29f){
+        // All devices with width >= 414
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:66]: I'd write it the other way around to have less comparisons: first check for `transform >= 1.29`, then for `transform > 1.0`

File: XBMC Remote/RemoteController.m [hunk 3]
```diff
@@ -142,20 +143,12 @@ - (void)configureView{
...
-            frame.size.height = frame.size.height *transform;
-            frame.size.width = frame.size.width*transform;
-            [remoteControlView setFrame:CGRectMake(frame.origin.x, frame.origin.y + 12, frame.size.width * 1.075, frame.size.height * 1.075)];
-        }
-        CGRect frame = subsInfoLabel.frame;
+        float transform = GET_TRANSFORM_X;
+        CGRect frame = remoteControlView.frame;
+        frame.size.height = frame.size.height *transform;
+        frame.size.width = frame.size.width*transform;
+        [remoteControlView setFrame:CGRectMake(frame.origin.x, frame.origin.y, frame.size.width* 1.05, frame.size.height* 1.05)];
```
Comment (kambala-decapitator) [inline XBMC Remote/RemoteController.m:150]: not clear what this magic multiplier means

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
Runtime test ok.

General code request: avoid using `float` (and `f` suffix in literals) when dealing with `CGFloat` values - on 64-bit devices (i.e. anything modern) it's resolved to `double`.

File: XBMC Remote/RemoteController.m (hunk 3 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m:150]: It was there before, with a different value. This is just scaling the image a bit further to maximize the screen usage. If you do not apply this "hand tailored" magic number, there is more unused spaced around the remote image.

File: XBMC Remote/RemoteController.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/RemoteController.m]: Good point, done with https://github.com/xbmc/Official-Kodi-Remote-iOS/pull/112/commits/fe4e56b5273e4a840d4391813a6060f0ac631854.

File: XBMC Remote/RemoteController.m (hunk 3 above)
Comment (kambala-decapitator) [reply XBMC Remote/RemoteController.m:150]: let's give it an understandable identifier then

Comment (wutschel) [reply XBMC Remote/RemoteController.m]: Here we go: https://github.com/xbmc/Official-Kodi-Remote-iOS/pull/112/commits/08114497235e9127e6676b3d9fce4f6ba049289c.

File: XBMC Remote/NowPlaying.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: Nice :) -> https://github.com/xbmc/Official-Kodi-Remote-iOS/pull/112/commits/9534a507798c9dea76578b635a38c6f646b1a8b0

File: XBMC Remote/NowPlaying.m [hunk 4]
```diff
@@ -383,14 +383,12 @@ - (UIImage*)imageWithBorderFromImage:(UIImage*)source{
...
-    CGFloat center_x = frame.origin.x + frame.size.width/2;
-    CGFloat center_y = frame.origin.y + frame.size.height/2;
     CGFloat transform_x = GET_TRANSFORM_X;
     CGFloat transform_y = GET_TRANSFORM_Y;
     frame.size.width *= transform_x;
     frame.size.height *= transform_x;
-    frame.origin.x = center_x*transform_x - frame.size.width/2;
-    frame.origin.y = center_y*transform_y - frame.size.height/2;
+    frame.origin.x = CGRectGetMidX(frame)*transform_x - frame.size.width/2;
+    frame.origin.y = CGRectGetMidY(frame)*transform_y - frame.size.height/2;
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:391]: at this point `frame` is already larger than the original, is it correct to get center after the transformation?

Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: Sorry, I was moving too quick with this change. Fixed this with the latest force-push.

**Comment by kambala-decapitator** [comment]
@wutschel 

> General code request: avoid using `float` (and `f` suffix in literals) when dealing with `CGFloat` values - on 64-bit devices (i.e. anything modern) it's resolved to `double`.

will you fix this in the PR or merge as is and you'll fix it globally in a separate PR?

**Comment by wutschel** [comment]
> will you fix this in your new code in the PR or merge as is and you'll fix it globally in a separate PR?

I am not 100% sure I understand the change you ask for. How shall I avoid this case for example when scaling a CGFloat?

Example: 
```
float transform = 1.05f;
frame.size.height = frame.size.height * transform;
frame.size.width = frame.size.width * 1.10f;
```

Is the expectation to cast this back and forth like this?
```
float transform = 1.05f;
frame.size.height = (CGFloat)((int)frame.size.height * transform);
frame.size.width = (CGFloat)((int)frame.size.width * 1.10f);
```

In general I think such rework is worth a dedicated set of commits which can be done after all the relevant functional changes went in.

**Comment by kambala-decapitator** [comment]
no, expectation is that you don't use `float` type unless necessary. Treat `CGFloat` as `double`:
```objc
CGFloat transform = 1.05;
frame.size.height = frame.size.height * transform;
frame.size.width = frame.size.width * 1.10;
```

File: XBMC Remote/ShowInfoViewController.m [hunk 5]
```diff
@@ -767,8 +767,8 @@ -(void)createInfo{
     clearLogoWidth = self.view.frame.size.width - 20.0f;
     clearLogoHeight = 116;
     float transform = GET_TRANSFORM_X;
```
Comment (kambala-decapitator) [inline XBMC Remote/ShowInfoViewController.m:769]: missed this one ;)

**Comment by wutschel** [comment]
Hope this is what you were searching for.

In fact, there are dozens of places where the current code uses `float` and `f` in combination with `CGFloat`. Especially in direct assignments and in lots of `CGRectMake`() calls.

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

**Comment by kambala-decapitator** [comment]
thanks!

> In fact, there are dozens of places where the current code uses `float` and `f` in combination with `CGFloat`. Especially in direct assignments and in lots of `CGRectMake`() calls.

yes, true, but let's not add even more such code :) This can be fixed in a separate cleanup PR.

File: XBMC Remote/ShowInfoViewController.m (hunk 5 above)
Comment (wutschel) [reply XBMC Remote/ShowInfoViewController.m:769]: This is just used for multiplication with an `int`.

**Comment by wutschel** [comment]
> yes, true, but let's not add even more such code :) This can be fixed in a separate cleanup PR.

Agreed.

**Pull request merged**